doc/userguide/rules/intro.rst

```diff
@@ -349,3 +349,74 @@ reassembled streams, TLS-, SSL-, SSH-, FTP- and dcerpc-buffers.
 
 Note that there are some exceptions, e.g. the ``http_raw_uri`` keyword.
 See :ref:`rules-http-uri-normalization` for more information.
+
+
+Rule's Types and Categorization
+-------------------------------
+
+Once parsed, Suricata rules are categorized for performance and further
+processing (as different rule types will be handled by specific modules).
+
+This categorization is done taking into consideration the type of keywords used.
+This will impact:
+
+  - scope
+  - how often/ when is the rule matched against traffic
+  - against what the rule matches
+
+The following table lists all Suricata signature types, and how they impact the
+aspects aforementioned.
+
+.. list-table:: Suricata Rule Types
+    :header-rows: 1
+
+    * - Type
+      - Scope
+      - Inspected
+      - Matches
+      - Keyword Examples
+    * - Decode Events Only
+      - Packet
+      - Per-packet basis
+      - Packets that are broken on an IP address level
+      - ---
+    * - Packet
+      - Packet
+      - Per-packet basis
+      - Packet-level info (e.g.: header info, ttl...)
+      - ---
+    * - IP Only
+      - Flow
+      - Once per direction
+      - On the flow, on IP address level
+      - ---
+    * - IP Only (negated address)
+      - Flow
+      - Once per direction
+      - On the flow, on IP address level (negated addressed)
+      - ---
```

Comment:
there is special handling for sigs that could be ip-only, but contain address negation. The radix based implementation doesn't support this, so there is a special "like ip-only" logic to make this sigs behave more or less the same as non-negated ip-only

Reply:
I see. Need to study the code to understand this properly.
ok so, questions:
1. does this information concern the user? 
2. does it make sense to mention it separately? or rather, can we explain why is it separately mentioned w/o going into the internals somehow?

Reply:
it is at least visible to the use in the engine analysis

Reply:
I think I would rather do something like 2. I want to offer a good insight on how Suricata actually handles things, because sometimes the low level details will impact difference in behaviors that might otherwise not make sense. (Not sure if my words here make enough sense xD )

Reply:
They do :)
Let's do that then!